ewsonprem_connector.py

```diff
@@ -943,12 +964,49 @@ def _get_http_error_details(self, r):
 
         return ""
 
+    def all_authentication_methods(self, config):
```

Comment:
Can this function name be changed to something more descriptive? Maybe `set_authentication_method` would be better?

Reply:
Changed the definition name to set_authentication_method